File: SBS_Backend/sbs/src/main/java/com/securebanking/sbs/model/Transaction.java
package com.securebanking.sbs.model;

import jakarta.persistence.*;
import org.springframework.data.annotation.CreatedBy;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedBy;
import org.springframework.data.annotation.LastModifiedDate;

import java.time.LocalDateTime;

@Entity
@Table(name = "transactions")
public class Transaction {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer transactionId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "senderId", referencedColumnName = "userId")
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "senderAcc", referencedColumnName = "accountId", nullable = false)
    private Account senderAcc;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "receiverAcc", referencedColumnName = "accountId", nullable = false)
    private Account receiverAcc;

    @Column(nullable = false)
    private String transactionType;

    @Column(nullable = false)
    private String amount;

    @CreatedBy
    @Column(name = "created_by")
    private String createdBy;

    @CreatedDate
    @Column(name = "created_time")
    private LocalDateTime createdtime;

    @LastModifiedBy
    @Column(name = "last_modified_by")
    private String lastModifiedBy;

    @LastModifiedDate
    @Column(name = "last_modified_time")
    private LocalDateTime lastModifiedtime;

    @Column(nullable = false)
    private String status;

    public Integer getTransactionId() {
        return transactionId;
    }

    public void setTransactionId(Integer transactionId) {
        this.transactionId = transactionId;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Account getReceiverAcc() {
        return receiverAcc;
    }

    public void setReceiverAcc(Account receiverAcc) {
        this.receiverAcc = receiverAcc;
    }

    public String getTransactionType() {
        return transactionType;
    }

    public void setTransactionType(String transactionType) {
        this.transactionType = transactionType;
    }

    public String getCreatedBy() {
        return createdBy;
    }

    public void setCreatedBy(String createdBy) {
        this.createdBy = createdBy;
    }

    public LocalDateTime getCreatedtime() {
        return createdtime;
    }

    public void setCreatedtime(LocalDateTime createdtime) {
        this.createdtime = createdtime;
    }

    public String getLastModifiedBy() {
        return lastModifiedBy;
    }

    public void setLastModifiedBy(String lastModifiedBy) {
        this.lastModifiedBy = lastModifiedBy;
    }

    public LocalDateTime getLastModifiedtime() {
        return lastModifiedtime;
    }

    public void setLastModifiedtime(LocalDateTime lastModifiedtime) {
        this.lastModifiedtime = lastModifiedtime;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public Account getSenderAcc() {
        return senderAcc;
    }

    public void setSenderAcc(Account senderAcc) {
        this.senderAcc = senderAcc;
    }

    public String getAmount() {
        return amount;
    }

    public void setAmount(String amount) {
        this.amount = amount;
    }
}

File: SBS_Frontend/src/app/services/transaction.ts
import { user } from './user'; // Adjust the import path as necessary
import { account } from './account'; // Adjust the import path as necessary

export class transaction {

  transactionId?: string;
  user?: user;
  senderAcc?: account;
  receiverAcc?: account;
  transactionType: string | undefined | null;
  amount?: string;
  createdBy?: string;
  createdtime?: Date;
  lastModifiedBy?: string;
  lastModifiedtime?: Date;
  status?: string;

  constructor(
    user: user,
    // senderAcc: account,
    // receiverAcc: account,
    // transactionType: string,
    // amount: string,
    // status: string
  ) {
    this.user = user;
    // this.senderAcc = senderAcc;
    // this.receiverAcc = receiverAcc;
    // this.transactionType = transactionType;
    // this.amount = amount;
    // this.status = status;
    // Optionally set the createdtime to the current time
    // this.createdtime = new Date();
  }
}

// export class Transaction {
//     transactionId?: string;
//     userId: string;
//     senderAcc: string; // Assuming the account is identified by a string
//     receiverAcc: string;
//     transactionType: string;
//     amount: string;
//     status: string;
  
//     constructor(
//       userId: string,
//       senderAcc: string,
//       receiverAcc: string,
//       transactionType: string,
//       amount: string,
//       status: string
//     ) {
//       this.userId = userId;
//       this.senderAcc = senderAcc;
//       this.receiverAcc = receiverAcc;
//       this.transactionType = transactionType;
//       this.amount = amount;
//       this.status = status;
//     }
//   }
  